api/server.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express')
+const server = express();
+const userDb = require('../data/helpers/userDb');
+server.use(express.json());
+
+function allCaps(req, res, next) {
+    req.body.name = req.body.name.toUpperCase();
+    next();
+}
+
+server.get('/user/:id', (req, res) => {
+    userDb.get(req.params.id)
+        .then(user => {
+            if(user) {
+                res.status(200).json(user)
+            } else {
+                res.status(404).json({ message: 'user not found' })
+            }
+        })
+        .catch(err => {
+            res.status(500).json({ message: 'The user ID can not be found', error: err })
+        })
+})
+
+server.get('/user/posts/:id', (req, res) => {
```

Comment:
Great job with your syntax and keeping consistency.  Allows your code to be very readable.